File: Basic_Math/armstrong_number/armstrongNumber.rb
def armstrong_number(n)
    digitCount = n.to_s.length
    result  = 0
    while ( n > 0 ) 
        lastDigit = n % 10
        result += lastDigit ** digitCount
        n = n / 10
    end
    return result
end

print "armstrong number :", armstrong_number(1634)

File: Basic_Math/armstrong_number/armstrongNumber.ts
const armstrongNumberTs = (n: number): number => {
    const digitCount = String(n).length;
    let result = 0;
    while (n > 0) {
        let lastDigit = n % 10;
        result += lastDigit ** digitCount;
        n = Math.floor(n / 10);
    }
    return result;
};

console.log("armstrong number :", armstrongNumberTs(1634));
